resolver/src/system_conf/unix.rs

```diff
@@ -0,0 +1,173 @@
+// Copyright 2015-2017 Benjamin Fry <benjaminfry@me.com>
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+//! System configuration loading
+//!
+//! This module is resposible for parsing and returning the configuration from
+//!  the host system. It will read from the default location on each operating
+//!  system, e.g. most Unixes have this written to `/etc/resolv.conf`
+
+use std::fs::File;
+use std::io;
+use std::env;
+use std::io::Read;
+use std::path::Path;
+use std::net::SocketAddr;
+use std::str::FromStr;
+use std::time::Duration;
+
+use resolv_conf;
+use hostname::get_hostname;
+
+use trust_dns_proto::rr::Name;
+use config::*;
+
+
+pub(crate) fn read_system_conf() -> io::Result<(ResolverConfig, ResolverOpts)> {
+    let mut config = read_resolv_conf("/etc/resolv.conf")?;
+    if let Ok(domain) = env::var("LOCALDOMAIN") {
+        // FIXME: do we really want to fail here?
+        config.0.set_domain(Name::from_str(&domain)?);
+    } else if let Some(name) = get_hostname() {
+        // FIXME: do we really want to fail here?
+        config.0.set_domain(Name::from_str(&name)?);
```

Comment:
Is this correct? The domain should probably be just the domain portion of the hostname, right?

Reply:
Also, we should double check the precedence here. The env var overruling the resolv.conf is correct, but I’m not clear on if hostname should only be used in the case of it not being soecified in the resolv.conf file.

That is, should hostname only be used when it’s not in the env and not in the resolv.conf?

Reply:
definitely this is not correct, `LOCALDOMAIN` is use to overridden the `search` keyword of a system's resolv.conf file, according to http://man7.org/linux/man-pages/man5/resolv.conf.5.html . and crate `resolv-conf` provide a method to use hostname in case of it not being specified in the `resolv.conf` file.

Reply:
i will fix this.